lib/widgets/draggable_scrollable_modal_bottom_sheet.dart

```diff
@@ -0,0 +1,119 @@
+              // _DragHandleLayer's SizedBox.height.
+              padding: const EdgeInsets.only(top: kMinInteractiveDimension),
+
+              child: Column(
+                // Extend DraggableScrollableSheet to full width so the whole
+                // sheet responds to drag/scroll uniformly. There's probably a
+                // better way to do this.
+                crossAxisAlignment: CrossAxisAlignment.stretch,
+
+                children: [
+                  Builder(builder: builder),
+                ],
+              ),
+            ),
+          );
+        });
+  }
+}
+
+class _DragHandleLayer extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    ColorScheme colorScheme = Theme.of(context).colorScheme;
+    return SizedBox(
+      // In the spec, this is expressed as 22 logical pixels of top/bottom
+      // padding on the drag handle:
+      //   https://m3.material.io/components/bottom-sheets/specs#e69f3dfb-e443-46ba-b4a8-aabc718cf335
+      // The drag handle is specified with height 4 logical pixels, so we can
+      // get the same result by vertically centering the handle in a box with
+      // height 22 + 4 + 22 = 48. We have another way to say 48 --
+      // kMinInteractiveDimension -- which is actually not a bad way to
+      // express it, since the feature was announced as "an optional drag
+      // handle with an accessible 48dp hit target":
+      //   https://m3.material.io/components/bottom-sheets/overview#2cce5bae-eb83-40b0-8e52-5d0cfaa9b795
+      // As a bonus, that constant is easy to use at the other layer in the
+      // Stack where we set the starting position of the sheet's content to
+      // avoid the drag handle.
+      height: kMinInteractiveDimension,
+
+      child: Center(
+        child: ClipRRect(
+          clipBehavior: Clip.hardEdge,
+          borderRadius: const BorderRadius.all(Radius.circular(2)),
+          child: SizedBox(
+            // height / width / color (including opacity) from this table:
+            //   https://m3.material.io/components/bottom-sheets/specs#7c093473-d9e1-48f3-9659-b75519c2a29d
+            height: 4,
+            width: 32,
+            child: ColoredBox(color: colorScheme.onSurfaceVariant.withOpacity(0.40)),
+          ),
+        )));
+  }
+}
+
+/// Show a modal bottom sheet that drags and scrolls to present lots of content.
+///
+/// Aims to follow Material 3's "bottom sheet" with a drag handle:
+///   https://m3.material.io/components/bottom-sheets/overview
+Future<T?> showDraggableScrollableModalBottomSheet<T>({
+  required BuildContext context,
+  required WidgetBuilder builder,
+}) {
+  return showModalBottomSheet<T>(
+    context: context,
+
+    // Clip.hardEdge looks bad; Clip.antiAliasWithSaveLayer looks pixel-perfect
+    // on my iPhone 13 Pro but is marked as "much slower":
+    //   https://api.flutter.dev/flutter/dart-ui/Clip.html
+    clipBehavior: Clip.antiAlias,
```

Comment:
Huh — on my Pixel 5, I can't actually spot a difference regardless of what value I put here. Even `Clip.none`. (And I even restarted, just in case I was missing something about whether hot reload would work; but still nothing changed.)

Reply:
This one took some extra steps for me to reproduce. Did you test by adding enough content in the sheet that it became scrollable, then as you scrolled, watch what happened as the content crossed the curved boundary of the sheet in the corner?

Reply:
Ahh, I see. No, I didn't ­— I just looked at the corners of the sheet itself, without anything scrolling into them.